content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,1441 @@
+
+```javascript
+// Wait for transaction on subscribed account and re-request account data
+client.on("transaction", async (transaction) => {
+  // Reference: https://xrpl.org/account_info.html
+  const accountInfoRequest = {
+    "command": "account_info",
+    "account": address,
+    "ledger_index": transaction.ledger_index
+  }
+
+  const accountInfoResponse = await client.request(accountInfoRequest)
+  const accountData = prepareAccountData(accountInfoResponse.result.account_data, reserve)
+  appWindow.webContents.send('update-account-data', accountData)
+
+  // Step 3 code additions - start
+  const transactions = prepareTxData([{tx: transaction.transaction}])
+  appWindow.webContents.send('update-transaction-data', transactions)
+  // Step 4 code additions - end
+  
+})
+```
+
+4. In `view/preload.js`, add the following code at the bottom of `exposeInMainWorld()`:
+
+```javascript
+onEnterAccountAddress: (address) => {
+  ipcRenderer.send('address-entered', address)
+},
+onUpdateAccountData: (callback) => {
+  ipcRenderer.on('update-account-data', callback)
+},
+
+// Step 4 code additions - start
+onUpdateTransactionData: (callback) => {
+  ipcRenderer.on('update-transaction-data', callback)
+}
+// Step 4 code additions - end
+```
+
+5. Modify `view/template.html` by adding a new fieldset below the ones that are already there:
+
+```html
+        ...
+        Close Time: <strong id="ledger-close-time"></strong><br/>
+    </fieldset>
+
+    <!-- Step 4 code additions - start -->
+    <fieldset>
+        <legend>Transactions:</legend>
+        <table id="tx-table">
+            <thead>
+            <tr>
+                <th>Confirmed</th>
+                <th>Type</th>
+                <th>From</th>
+                <th>To</th>
+                <th>Value Delivered</th>
+                <th>Hash</th>
+            </tr>
+            </thead>
+            <tbody></tbody>
+        </table>
+    </fieldset>
+    <!-- Step 5 code additions - end -->
+
+    <dialog id="account-address-dialog">
+        <form method="dialog">
+```
+
+The table here will be filled dynamically with the accounts transactions.
+
+6. Add the following code at the bottom of `view/renderer.js`:
+
+```javascript
+const txTableBodyEl = document.getElementById('tx-table').tBodies[0]
+window.testEl = txTableBodyEl
+
+window.electronAPI.onUpdateTransactionData((_event, transactions) => {
+    for (let transaction of transactions) {
+        txTableBodyEl.insertAdjacentHTML( 'beforeend',
+        "<tr>" +
+            "<td>" + transaction.confirmed + "</td>" +
+            "<td>" + transaction.type + "</td>" +
+            "<td>" + transaction.from + "</td>" +
+            "<td>" + transaction.to + "</td>" +
+            "<td>" + transaction.value + "</td>" +
+            "<td>" + transaction.hash + "</td>" +
+            "</tr>"
+        )
+    }
+})
+```
+
+If you have come this far - congrats. Run the application by executing:
+
+```console
+npm run start
```

Comment:
Following all the steps to this point, when I type in my testnet address, nothing seems to happen. (The previous section step 3 was displaying my account information correctly)

I'm unsure why. I notice there are files in the "All files you'll need" at the beginning which are never mentioned in these instructions. Ex. `tx-history.html` is never mentioned and 4_tx-history.js` is also never mentioned. Will try to debug later.

Reply:
Re-running through the steps still ending up in the same situation, this is the zip file I get after going through the steps to this point: 
[test3.zip](https://github.com/XRPLF/xrpl-dev-portal/files/12034891/test3.zip)

Reply:
Thanks for the Snapshot! So I found this one in the Dev Tools:

<img width="392" alt="image" src="https://github.com/XRPLF/xrpl-dev-portal/assets/102560752/05bba92f-6546-45e2-83c2-61e7fa413035">

Reply:
Which means I will add a section on how to access the Dev Tools somewhere at the beginning of the tutorial. Sometimes when you spend a long time with something you take things for granted which one shouldn't...

Reply:
Ok, so the function got lost in the preloader. I added the whole block to the tutorial to prevent misunderstandings.

Reply:
@JST5000 I'll leave that unresolved as I suppose you can't see the comments if I resolve the conversation.

Reply:
In your case, this would have to be re-added in preload.js:

```javascript
onUpdateLedgerData: (callback) => {
        ipcRenderer.on('update-ledger-data', callback)
    },
```

Reply:
Developer Tools Tip has been added just after the instruction on how to start the "Hello world!" application

Reply:
> Which means I will add a section on how to access the Dev Tools somewhere at the beginning of the tutorial. Sometimes when you spend a long time with something you take things for granted which one shouldn't...

Ah, I didn't realize Electron had dev tools - I wasn't thinking of it like a browser - oops!

Reply:
Then I can close this :)